feat: Added a customizable modal which double checks with the user

dhruvkanth opened the pull request:
**Closes:** #95 

**What was done:**
- Added a modal which confirms with the user if they want to proceed
**How to test:**
- Create a state to declare whether the modal is open or closed
- And can use the ConfirmModal component by passing in the Header, Paragraph Text, Accept Button text, Cancel button text, Modal State Function, Function to call on clicking accept

ejiang25 committed:
feat: created generalized button component

ejiang25 committed:
feat: changed logout button

ejiang25 committed:
feat: changed new meeting button

ejiang25 committed:
feat: changed submit button

ejiang25 committed:
feat: changed after meeting buttons

ejiang25 committed:
feat: changed cancel button for new meeting

ejiang25 committed:
Merge branch 'main' into AEYB/#93-make-generalised-button

ejiang25 committed:
feat: changed ActiveMeeting buttons

ejiang25 committed:
feat: changed all buttons to Button component

ejiang25 committed:
Merge branch 'main' into AEYB/#93-make-generalised-button

ejiang25 committed:
Merge branch 'main' into AEYB/#93-make-generalised-button

dhruvkanth committed:
feat: Added a customizable modal which double checks with the user

dhruvkanth committed:
feat: Added a customizable modal which double checks with the user

unix2dossss committed:
added generalized button animation

dhruvkanth committed:
style: Changed the style of the ConfirmModal.tsx

dhruvkanth committed:
style: Changed the style of the ConfirmModal.tsx

ejiang25 committed:
feat: hide past meetings

ejiang25 committed:
Merge branch 'main' into AEYB/#133-calendar-view-needs-to-be-changed-to-monthly-view

ejiang25 committed:
feat: monthly view of meetings

ejiang25 committed:
feat: sort meetings

ejiang25 committed:
feat: included meetings during last day of month

TylerDonghwi committed:
Merge branch 'main' into AEYB/#95-confirm-popup

TylerDonghwi commented:
I want you to add this component to two features in the app. 
First is when you try to submit the creation of a new meeting, can you make it such that it will show before creating the meeting? 
![image](https://github.com/UoaWDCC/aeyb-monitor/assets/98932343/7c7fa5f2-222c-419e-b107-5424b8453ca8)
Second is replacing this
![image](https://github.com/UoaWDCC/aeyb-monitor/assets/98932343/ca294e9f-7d6f-41a0-b638-23839cd39577)

- maybe you can also create a reject modal which would show when the inputs are in the wrong format? Currenlty we are using alert("error message") currently but since you made the confirm, may as well create the reject modal as well. Let me know what you think.

TylerDonghwi committed:
refactor weeklyinstance

TylerDonghwi committed:
Merge pull request #140 from UoaWDCC/buton-animations

Buton animations

ejiang25 committed:
refactored the code

ejiang25 committed:
added missing destructure

TylerDonghwi committed:
Update CalendarPage.tsx - removed react fragment

TylerDonghwi committed:
Merge pull request #139 from UoaWDCC/AEYB/#133-calendar-view-needs-to-be-changed-to-monthly-view

Aeyb/#133 calendar view needs to be changed to monthly view

richard-ch committed:
Aeyb/#134 end meeting endpoint be (#143)

* Added endmeeting request type and new endpoint

* added endmeeting controller and new routes

* removed testing comments

jacobatan committed:
Fix infinite loading

jacobatan committed:
Edited how context works

jacobatan committed:
Removed unused code

TylerDonghwi committed:
Merge pull request #146 from UoaWDCC/AEYB/#145-infinite-loading

Fix infinite loading

unix2dossss committed:
added generalised button component: can change button size and colour

unix2dossss committed:
allows user to enter hex colors, and dynamically sets text color depending on bg

jacobatan committed:
Initial changes

jacobatan committed:
Added end meeting api call

jacobatan committed:
Removed testing code

jacobatan committed:
Code restructe of active meeting page

jacobatan committed:
Changed PR template

jacobatan committed:
Removed delay variable

jacobatan committed:
PR Comments

tlooi committed:
Merge pull request #147 from UoaWDCC/AEYB/#135-end-meeting

Aeyb/#135 end meeting

unix2dossss committed:
Changed a few buttons over the website to the general buttons. Added extraStyle prop.

unix2dossss committed:
Not Found Page. ~ Completed

unix2dossss committed:
Addressed review feedback

Addressed PR code review feedback

unix2dossss committed:
Decreased Logo Size. Refactored Code.

Logo looks less overwhelming. Refactored code to fit React conventions.

TylerDonghwi committed:
Merge pull request #148 from UoaWDCC/Aeyb/#127-404-not-found-page

Not Found Page

unix2dossss committed:
Customized Post Meeting Button. Correctly Installed TinyColor2

unix2dossss committed:
added textColor prop, Refactored: Submit & Close Button (new-meeting pop up page)

unix2dossss committed:
Refactored After Meeting Buttons with General Button Component

unix2dossss committed:
Added Custom Sizing Prop for General Button. Refactored Settings Page Buttons.

unix2dossss committed:
Refactored Calendar Buttons to Generalised Buttons

unix2dossss committed:
removed commented legacy code

unix2dossss committed:
Merge branch 'main' into AEYB/#93-make-generalised-button

unix2dossss committed:
Random Fix

TylerDonghwi committed:
Merge pull request #131 from UoaWDCC/AEYB/#93-make-generalised-button

Aeyb/#93 make generalised button

unix2dossss committed:
Basic Scroll Component Created.

unix2dossss committed:
Added Dynamic Scroll Button Visibility

unix2dossss committed:
Aeyb/#144 Go to Top Button (#149)

* Basic Scroll Component Created.

* Added Dynamic Scroll Button Visibility

tlooi committed:
initial tabs

tlooi committed:
fix scrolling pages on default

tlooi committed:
add user page with roles

tlooi committed:
add check to ensure don't remove admin or default roles

tlooi committed:
add removing roles on client

tlooi committed:
prevent tab content body from overflowing

tlooi committed:
feat: add roles

tlooi committed:
lots of nested tabs :D

tlooi committed:
add mui popup

jacobatan committed:
Added new query method

jacobatan committed:
Updated remove method to do logic on DB

tlooi committed:
feat: implement permission modification on roles

Co-authored-by: jacobatan <jacob.tan229@gmail.com>

tlooi committed:
refactor: move tabs into separate component

tlooi committed:
refactor: extract out ViewRolesWindow and ViewPermissions component

tlooi committed:
refactor: moved out UserRoleRow

tlooi committed:
refactor: separate out ViewUserWindow

tlooi committed:
refactor: cleanup code imports

tlooi committed:
style: fix + button centering and add fake profile component

tlooi committed:
feat: add role button disappears if no more to add

tlooi committed:
style: prevent select of text

tlooi committed:
feat: create new roles and assign permissions

tlooi committed:
fix: color not generating properly

tlooi committed:
refactor: clean up imports

tlooi committed:
refactor: remove unused folders and removed commented lines

jacobatan committed:
Minor changes

tlooi committed:
allow customizable tabs

tlooi committed:
refactor: implement review changes

tlooi committed:
Merge branch 'AEYB/#132-Assign-roles-to-users' of https://github.com/UoaWDCC/aeyb-monitor into AEYB/#132-Assign-roles-to-users

tlooi committed:
refactor: added implicit return

tlooi committed:
style: fixed tab styling

tlooi committed:
fix: potential error by modifying JSX return to JSX element

- Prevents tabManager take in params not props to prevent accidental misuse and potential breakage of app via more hooks than prior render

tlooi committed:
revert: revert to old

jacobatan committed:
Improved code readability

tlooi committed:
feat: delete roles from database

tlooi committed:
added loaders for data

jacobatan committed:
Merge pull request #150 from UoaWDCC/AEYB/#132-Assign-roles-to-users

Aeyb/#132 assign roles to users

dhruvkanth committed:
style: fixed button styling

dhruvkanth committed:
Merge remote-tracking branch 'origin/AEYB/#95-confirm-popup' into AEYB/#95-confirm-popup

# Conflicts:
#	frontend/src/utility_components/ConfirmModal/ConfirmModal.tsx

dhruvkanth committed:
style: fixed button styling

dhruvkanth committed:
feat: added double check for creating new meetings

jacobatan reviewed (approved):
lgtm
